In [48]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn import set_config; set_config(display='diagram')
from scipy import stats
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import MinMaxScaler, StandardScaler
from sklearn.pipeline import Pipeline, make_pipeline
from sklearn.compose import ColumnTransformer, make_column_transformer, make_column_selector
from sklearn.impute import SimpleImputer, KNNImputer
from sklearn.preprocessing import RobustScaler, OneHotEncoder, OrdinalEncoder
from sklearn.metrics import make_scorer
from sklearn.model_selection import cross_val_score
from sklearn.neighbors import KNeighborsRegressor
from sklearn.linear_model import Ridge, Lasso, LinearRegression
from sklearn.svm import SVR
from sklearn.model_selection import GridSearchCV, RandomizedSearchCV
from sklearn.ensemble import AdaBoostRegressor
from sklearn.tree import DecisionTreeRegressor
from sklearn.ensemble import VotingRegressor
from sklearn.ensemble import GradientBoostingRegressor
from sklearn.ensemble import StackingRegressor
from sklearn.ensemble import RandomForestRegressor
from xgboost import XGBRegressor
from sklearn.model_selection import RepeatedKFold
from numpy import mean
from numpy import std

In [49]:
#dataset_d16 = '../raw_data/preproc_data_d16.csv'
dataset_d2 = '../raw_data/preproc_data_d2.csv'
#data_d16 = pd.read_csv(dataset_d16)
data_d2 = pd.read_csv(dataset_d2)
#data_d16.drop(columns='date', inplace = True)
data_d2.drop(columns='date', inplace=True)
#data_d16.drop(columns='jour', inplace = True)
data_d2.drop(columns='jour', inplace=True)
#data_d16.head()
data_d2

service   CA_TTC  moyen_7_services  moyen_31_services  \
0       soir   8345.0               NaN                NaN   
1       soir   9045.5               NaN                NaN   
2       midi   3939.0               NaN                NaN   
3       soir  10024.0               NaN                NaN   
4       midi   4836.0               NaN                NaN   
...      ...      ...               ...                ...   
1229    soir   7230.0       7909.714286        8088.625806   
1230    midi   2104.0       2849.357143        3314.595484   
1231    soir   9816.5       7880.642857        8008.254839   
1232    soir   8160.5       8161.428571        7883.061290   
1233    midi   1202.5       2846.428571        3108.821290   

      moyenne_3der_j&service  match_edf  roland_garros  fashion_week  \
0                        NaN          0              0             0   
1                        NaN          0              0             0   
2                        NaN          0              0             0   
3                        NaN          0              0             0   
4                        NaN          0              0             0   
...                      ...        ...            ...           ...   
1229                 8803.83          0              0             0   
1230                 3076.17          0              0             0   
1231                 8632.17          0              0             0   
1232                 8903.67          0              0             0   
1233                 3432.42          0              0             0   

      match_happening  match_happening_cl  ...  cos_jour_de_sem  \
0                 0.0                 0.0  ...        -0.222521   
1                 0.0                 0.0  ...        -0.900969   
2                 0.0                 0.0  ...        -0.900969   
3                 0.0                 0.0  ...        -0.900969   
4                 0.0                 0.0  ...        -0.900969   
...               ...                 ...  ...              ...   
1229              0.0                 0.0  ...        -0.900969   
1230              0.0                 0.0  ...        -0.222521   
1231              0.0                 0.0  ...        -0.222521   
1232              0.0                 0.0  ...         0.623490   
1233              0.0                 0.0  ...         0.623490   

      sin_jour_de_sem  cos_jour_du_mois  sin_jour_du_mois  cos_mois_de_annee  \
0            0.974928          0.979530      2.012985e-01           0.866025   
1            0.433884          0.918958      3.943559e-01           0.866025   
2            0.433884          0.918958      3.943559e-01           0.866025   
3           -0.433884          0.820763      5.712682e-01           0.866025   
4           -0.433884          0.820763      5.712682e-01           0.866025   
...               ...               ...               ...                ...   
1229        -0.433884          0.918958     -3.943559e-01          -0.866025   
1230        -0.974928          0.979530     -2.012985e-01          -0.866025   
1231        -0.974928          0.979530     -2.012985e-01          -0.866025   
1232        -0.781831          1.000000     -2.449294e-16          -0.866025   
1233        -0.781831          1.000000     -2.449294e-16          -0.866025   

      sin_mois_de_annee  cos_jour_annee sin_jour_annee cos_semaine_annee  \
0                   0.5        0.999852       0.017213          0.992709   
1                   0.5        0.999407       0.034422          0.992709   
2                   0.5        0.999407       0.034422          0.992709   
3                   0.5        0.998667       0.051620          0.992709   
4                   0.5        0.998667       0.051620          0.992709   
...                 ...             ...            ...               ...   
1229               -0.5       -0.890028      -0.455907         -0.885456   
1230               -0

In [50]:
X_d2 = data_d2.drop(columns=["CA_TTC"])
y_d2 = data_d2.CA_TTC
#X_d16 = data_d2.drop(columns=["CA_TTC"])
#y_d16 = data_d2.CA_TTC
X_d2.columns


Index(['service', 'moyen_7_services', 'moyen_31_services',
       'moyenne_3der_j&service', 'match_edf', 'roland_garros', 'fashion_week',
       'match_happening', 'match_happening_cl', 'temp', 'feels_like',
       'temp_min', 'temp_max', 'wind_speed', 'clouds_all', 'vacances_paris',
       'clear', 'clouds', 'drizzle', 'drizzle_and_rain', 'fog', 'mist', 'rain',
       'thunderstorm', 'cos_jour_de_sem', 'sin_jour_de_sem',
       'cos_jour_du_mois', 'sin_jour_du_mois', 'cos_mois_de_annee',
       'sin_mois_de_annee', 'cos_jour_annee', 'sin_jour_annee',
       'cos_semaine_annee', 'sin_semaine_annee'],
      dtype='object')

In [51]:
categorical_features = ['service', 'match_happening', 'match_happening_cl', 'match_edf', 'roland_garros', 'fashion_week','vacances_paris']
cyclical_features = ['cos_jour_de_sem','sin_jour_de_sem', 'cos_jour_du_mois', 'sin_jour_du_mois','cos_mois_de_annee', 'sin_mois_de_annee', 'cos_jour_annee','sin_jour_annee', 'cos_semaine_annee', 'sin_semaine_annee']
numerical_features = ['moyen_7_services','moyen_31_services','moyenne_3der_j&service','temp', 'feels_like', 'temp_min', 'temp_max', 'wind_speed', 'clouds_all']
ordinal_features =['clear', 'clouds', 'drizzle', 'drizzle_and_rain', 'fog', 'mist', 'rain', 'thunderstorm']


In [52]:
feat_ordinal_dict = {
    'clear': ['0',0,'sky is clear'],
    'clouds': ['0',0,'scattered clouds','few clouds','broken clouds','overcast clouds'],
    'drizzle': ['0',0,'light intensity drizzle','drizzle','heavy intensity drizzle'],
    'drizzle_and_rain': ['0',0,'light intensity drizzle rain','rain and drizzle'],
    'fog': ['0',0,'fog'],
    'mist': ['0',0,'mist'],
    'rain': ['0',0,'light rain','light intensity shower rain','moderate rain','heavy intensity rain'],
    'thunderstorm': ['0',0,'proximity thunderstorm','thunderstorm','thunderstorm with light rain','thunderstorm with heavy rain']
}

feat_ordinal = sorted(feat_ordinal_dict.keys()) # sort alphabetically
feat_ordinal_values_sorted = [feat_ordinal_dict[i] for i in feat_ordinal]

encoder_ordinal = OrdinalEncoder(
    categories=feat_ordinal_values_sorted,
    dtype= np.int64,
    handle_unknown="use_encoded_value",
    unknown_value=-1 # Considers unknown values as worse than "missing"
)

preproc_ordinal = make_pipeline(
    SimpleImputer(strategy="constant", fill_value="missing"),
    encoder_ordinal,
    RobustScaler()
)

preproc_ordinal

Pipeline(steps=[('simpleimputer',
                 SimpleImputer(fill_value='missing', strategy='constant')),
                ('ordinalencoder',
                 OrdinalEncoder(categories=[['0', 0, 'sky is clear'],
                                            ['0', 0, 'scattered clouds',
                                             'few clouds', 'broken clouds',
                                             'overcast clouds'],
                                            ['0', 0, 'light intensity drizzle',
                                             'drizzle',
                                             'heavy intensity drizzle'],
                                            ['0', 0,
                                             'light intensity drizzle rain',
                                             'rain and drizzle'],
                                            ['0', 0, 'fog'], ['0', 0, 'mist'],
                                            ['0', 0, 'light rain',
                                             'light intensity shower rain',
                                             'moderate rain',
                                             'heavy intensity rain'],
                                            ['0', 0, 'proximity thunderstorm',
                                             'thunderstorm',
                                             'thunderstorm with light rain',
                                             'thunderstorm with heavy rain']],
                                dtype=<class 'numpy.int64'>,
                                handle_unknown='use_encoded_value',
                                unknown_value=-1)),
                ('robustscaler', RobustScaler())])

In [53]:
preproc_categorical = make_pipeline(
    SimpleImputer(strategy="constant", fill_value="missing"),
    OneHotEncoder(handle_unknown="ignore"))

preproc_cyclical = make_pipeline(
    KNNImputer())

preproc_numerical = make_pipeline(
    KNNImputer(),
    RobustScaler())

preproc = make_column_transformer(
    (preproc_categorical, categorical_features),
    (preproc_cyclical, cyclical_features),
    (preproc_numerical, numerical_features),
    (preproc_ordinal, ordinal_features),
    remainder ="drop")

pipe = make_pipeline(
    preproc,
)

In [54]:
pipe

Pipeline(steps=[('columntransformer',
                 ColumnTransformer(transformers=[('pipeline-1',
                                                  Pipeline(steps=[('simpleimputer',
                                                                   SimpleImputer(fill_value='missing',
                                                                                 strategy='constant')),
                                                                  ('onehotencoder',
                                                                   OneHotEncoder(handle_unknown='ignore'))]),
                                                  ['service', 'match_happening',
                                                   'match_happening_cl',
                                                   'match_edf', 'roland_garros',
                                                   'fashion_week',
                                                   'vacances_paris']),
                                                 ('pipe...
                                                                                               'rain',
                                                                                               'heavy '
                                                                                               'intensity '
                                                                                               'rain'],
                                                                                              ['0',
                                                                                               0,
                                                                                               'proximity '
                                                                                               'thunderstorm',
                                                                                               'thunderstorm',
                                                                                               'thunderstorm '
                                                                                               'with '
                                                                                               'light '
                                                                                               'rain',
                                                                                               'thunderstorm '
                                                                                               'with '
                                                                                               'heavy '
                                                                                               'rain']],
                                                                                  dtype=<class 'numpy.int64'>,
                                                                                  handle_unknown='use_encoded_value',
                                                                                  unknown_value=-1)),
                                                                  ('robustscaler',
                                                                   RobustScaler())]),
                                                  ['clear', 'clouds', 'drizzle',
                                                   'drizzle_and_rain', 'fog',
                                                   'mist', 'rain',
                                                   'thunderstorm'])]))])

In [55]:
run_test = True
if run_test == True:
  XGB = XGBRegressor(objective='reg:squarederror', 
                      max_depth=4, 
                      n_estimators=40, #first param
                      learning_rate=0.1)   #second param
              
  pipe_xgb = make_pipeline(preproc, XGB)
  cv = RepeatedKFold(n_splits=5, n_repeats=2, random_state=0)  #fourth param
  n_scores = cross_val_score(pipe_xgb, X_d2, y_d2, cv=cv, scoring='neg_mean_absolute_error', n_jobs=-1)
  print('MAE: %.3f (%.3f)' % (mean(n_scores), std(n_scores)))

MAE: -939.086 (51.364)


In [56]:
ohe = OneHotEncoder(sparse=False)
ohe.fit(X_d2[['service']])
X_d2['service'] = ohe.transform(X_d2[['service']])
X_d2.columns

Index(['service', 'moyen_7_services', 'moyen_31_services',
       'moyenne_3der_j&service', 'match_edf', 'roland_garros', 'fashion_week',
       'match_happening', 'match_happening_cl', 'temp', 'feels_like',
       'temp_min', 'temp_max', 'wind_speed', 'clouds_all', 'vacances_paris',
       'clear', 'clouds', 'drizzle', 'drizzle_and_rain', 'fog', 'mist', 'rain',
       'thunderstorm', 'cos_jour_de_sem', 'sin_jour_de_sem',
       'cos_jour_du_mois', 'sin_jour_du_mois', 'cos_mois_de_annee',
       'sin_mois_de_annee', 'cos_jour_annee', 'sin_jour_annee',
       'cos_semaine_annee', 'sin_semaine_annee'],
      dtype='object')

In [57]:
X_d2.columns

Index(['service', 'moyen_7_services', 'moyen_31_services',
       'moyenne_3der_j&service', 'match_edf', 'roland_garros', 'fashion_week',
       'match_happening', 'match_happening_cl', 'temp', 'feels_like',
       'temp_min', 'temp_max', 'wind_speed', 'clouds_all', 'vacances_paris',
       'clear', 'clouds', 'drizzle', 'drizzle_and_rain', 'fog', 'mist', 'rain',
       'thunderstorm', 'cos_jour_de_sem', 'sin_jour_de_sem',
       'cos_jour_du_mois', 'sin_jour_du_mois', 'cos_mois_de_annee',
       'sin_mois_de_annee', 'cos_jour_annee', 'sin_jour_annee',
       'cos_semaine_annee', 'sin_semaine_annee'],
      dtype='object')

In [58]:
weather_encoded = (encoder_ordinal.fit_transform(X_d2[['clear','clouds', 'drizzle', 'drizzle_and_rain', 'fog', 'mist', 'rain', 'thunderstorm']]))
weather_encoded_df = pd.DataFrame(weather_encoded)

In [59]:
weather_encoded_df[0].dtypes

dtype('int64')

In [60]:
weather_encoded_df[0]

0       0
1       0
2       0
3       0
4       0
       ..
1229    2
1230    0
1231    2
1232    0
1233    2
Name: 0, Length: 1234, dtype: int64

In [61]:
X_d2['clear'] = weather_encoded_df[0]
X_d2['clouds'] = weather_encoded_df[1]
X_d2['drizzle'] = weather_encoded_df[2]
X_d2['drizzle_and_rain'] = weather_encoded_df[3]
X_d2['fog'] = weather_encoded_df[4]
X_d2['mist'] = weather_encoded_df[5]
X_d2['rain'] = weather_encoded_df[6]
X_d2['thunderstorm'] = weather_encoded_df[7]

In [62]:
num_encoded = preproc_numerical.fit_transform(X_d2[numerical_features])
num_encoded_df = pd.DataFrame(num_encoded)

In [63]:
X_d2['moyen_7_services'] = num_encoded_df[0]
X_d2['moyen_31_services'] = num_encoded_df[1]
X_d2['moyenne_3der_j&service'] = num_encoded_df[2]
X_d2['temp'] = num_encoded_df[3]
X_d2['feels_like'] = num_encoded_df[4]
X_d2['temp_min'] = num_encoded_df[5]
X_d2['temp_max'] = num_encoded_df[6]
X_d2['wind_speed'] = num_encoded_df[7]
X_d2['clouds_all'] = num_encoded_df[8]

In [64]:
X_train,X_test, y_train, y_test = train_test_split(X_d2, y_d2)

In [65]:
X_train.dtypes

service                   float64
moyen_7_services          float64
moyen_31_services         float64
moyenne_3der_j&service    float64
match_edf                   int64
roland_garros               int64
fashion_week                int64
match_happening           float64
match_happening_cl        float64
temp                      float64
feels_like                float64
temp_min                  float64
temp_max                  float64
wind_speed                float64
clouds_all                float64
vacances_paris              int64
clear                       int64
clouds                      int64
drizzle                     int64
drizzle_and_rain            int64
fog                         int64
mist                        int64
rain                        int64
thunderstorm                int64
cos_jour_de_sem           float64
sin_jour_de_sem           float64
cos_jour_du_mois          float64
sin_jour_du_mois          float64
cos_mois_de_annee         float64
sin_mois_de_an

In [79]:
X_train.describe()

service  moyen_7_services  moyen_31_services  \
count  925.000000        925.000000         925.000000   
mean     0.500541          0.248113           0.214527   
std      0.500270          0.530115           0.504601   
min      0.000000         -0.601293          -0.517842   
25%      0.000000         -0.214212          -0.212246   
50%      1.000000         -0.002897          -0.009554   
75%      1.000000          0.784175           0.791244   
max      1.000000          1.319723           1.053460   

       moyenne_3der_j&service   match_edf  roland_garros  fashion_week  \
count              925.000000  925.000000     925.000000    925.000000   
mean                 0.200355    0.015135       0.073514      0.101622   
std                  0.605407    0.122156       0.261119      0.302314   
min                 -0.698942    0.000000       0.000000      0.000000   
25%                 -0.327535    0.000000       0.000000      0.000000   
50%                  0.012933    0.000000       0.000000      0.000000   
75%                  0.701560    0.000000       0.000000      0.000000   
max                  1.550768    1.000000       1.000000      1.000000   

       match_happening  match_happening_cl        temp  ...  cos_jour_de_sem  \
count            925.0               925.0  925.000000  ...       925.000000   
mean               0.0                 0.0    0.043258  ...        -0.013274   
std                0.0                 0.0    0.648541  ...         0.706304   
min                0.0                 0.0   -1.371843  ...        -0.900969   
25%                0.0                 0.0   -0.474120  ...        -0.900969   
50%                0.0                 0.0    0.021118  ...        -0.222521   
75%                0.0                 0.0    0.545342  ...         0.623490   
max                0.0                 0.0    1.958178  ...         1.000000   

       sin_jour_de_sem  cos_jour_du_mois  sin_jour_du_mois  cos_mois_de_annee  \
count     9.250000e+02        925.000000      9.250000e+02       9.250000e+02   
mean     -1.676819e-02         -0.048580     -4.498354e-03      -9.254607e-02   
std       7.083498e-01          0.700929      7.123183e-01       6.950097e-01   
min      -9.749279e-01         -0.994869     -9.987165e-01      -1.000000e+00   
25%      -7.818315e-01         -0.758758     -7.247928e-01      -8.660254e-01   
50%      -2.449294e-16         -0.050649     -2.449294e-16      -1.836970e-16   
75%       7.818315e-01          0.688967      7.247928e-01       5.000000e-01   
max       9.749279e-01          1.000000      9.987165e-01       1.000000e+00   

       sin_mois_de_annee  cos_jour_annee  sin_jour_annee  cos_semaine_annee  \
count       9.250000e+02      925.000000      925.000000         925.000000   
mean       -1.038870e-01       -0.110801       -0.081063          -0.100720   
std         7.061625e-01        0.724315        0.676447           0.716077   
min        -1.000000e+00       -0.999963       -0.999991          -1.000000   
25%        -8.660254e-01       -0.873807       -0.769415          -0.822984   
50%        -2.449294e-16       -0.200891       -0.077386          -0.120537   
75%         5.000000e-01        0.598181        0.538005           0.568065   
max         1.000000e+00        1.000000        0.999917           1.000000   

       sin_semaine_annee  
count         925.000000  
mean           -0.085556  
std             0.686173  
min            -1.000000  
25%            -0.748511  
50%            -0.120537  
75%             0.568065  
max             1.000000  

[8 rows x 34 columns]

In [67]:
X_train.columns

Index(['service', 'moyen_7_services', 'moyen_31_services',
       'moyenne_3der_j&service', 'match_edf', 'roland_garros', 'fashion_week',
       'match_happening', 'match_happening_cl', 'temp', 'feels_like',
       'temp_min', 'temp_max', 'wind_speed', 'clouds_all', 'vacances_paris',
       'clear', 'clouds', 'drizzle', 'drizzle_and_rain', 'fog', 'mist', 'rain',
       'thunderstorm', 'cos_jour_de_sem', 'sin_jour_de_sem',
       'cos_jour_du_mois', 'sin_jour_du_mois', 'cos_mois_de_annee',
       'sin_mois_de_annee', 'cos_jour_annee', 'sin_jour_annee',
       'cos_semaine_annee', 'sin_semaine_annee'],
      dtype='object')

In [74]:
#X_train.drop(columns=['thunderstorm','fog','drizzle','drizzle_and_rain','drizzle','clear','match_happening_cl','match_happening','match_edf']

service  match_edf  roland_garros  fashion_week  match_happening  \
25        1.0          0              0             0              0.0   
208       0.0          0              0             0              0.0   
35        0.0          0              0             1              0.0   
783       0.0          0              0             0              0.0   
997       0.0          0              0             0              0.0   
...       ...        ...            ...           ...              ...   
512       0.0          0              0             0              0.0   
638       0.0          0              0             0              0.0   
1024      0.0          0              0             0              0.0   
1165      1.0          0              0             1              0.0   
561       1.0          0              0             0              0.0   

      match_happening_cl      temp  feels_like  temp_min  temp_max  ...  \
25                   0.0 -0.595031   -0.627948 -0.541632 -0.642169  ...   
208                  0.0 -0.286957   -0.298117 -0.258862 -0.336948  ...   
35                   0.0 -1.207039   -1.303469 -1.138500 -1.222088  ...   
783                  0.0 -0.687785   -0.623191 -0.615829 -0.695984  ...   
997                  0.0  0.582609    0.567691  0.603462  0.548193  ...   
...                  ...       ...         ...       ...       ...  ...   
512                  0.0  0.568530    0.573241  0.532564  0.558635  ...   
638                  0.0 -1.052174   -1.254311 -1.000000 -1.041365  ...   
1024                 0.0  0.723395    0.728642  0.539984  0.826908  ...   
1165                 0.0  0.161905    0.155401  0.236603  0.152209  ...   
561                  0.0  0.142029    0.145094  0.198681  0.132932  ...   

      cos_jour_de_sem  sin_jour_de_sem  cos_jour_du_mois  sin_jour_du_mois  \
25           0.623490         0.781831         -0.954139          0.299363   
208         -0.222521         0.974928         -0.994869         -0.101168   
35           0.623490        -0.781831         -0.758758         -0.651372   
783         -0.222521         0.974928          0.688967          0.724793   
997          0.623490        -0.781831         -0.250653         -0.968077   
...               ...              ...               ...               ...   
512          0.623490         0.781831         -0.994869         -0.101168   
638         -0.900969         0.433884         -0.612106         -0.790776   
1024        -0.222521        -0.974928          0.688967          0.724793   
1165        -0.900969        -0.433884          0.151428         -0.988468   
561         -0.222521        -0.974928         -0.612106          0.790776   

      cos_mois_de_annee  sin_mois_de_annee  cos_jour_annee  sin_jour_annee  \
25         8.660254e-01       5.000000e-01        0.971100        0.238673   
208       -5.000000e-01       8.660254e-01       -0.251190        0.967938   
35         8.660254e-01       5.000000e-01        0.946988        0.321270   
783        5.000000e-01       8.660254e-01        0.823923        0.566702   
997       -5.000000e-01      -8.660254e-01       -0.618671       -0.785650   
...                 ...                ...             ...             ...   
512       -1.836970e-16      -1.000000e+00       -0.251190       -0.967938   
638        8.660254e-01      -5.000000e-01        0.761104       -0.648630   
1024      -1.836970e-16      -1.000000e+00       -0.428892       -0.903356   
1165      -1.000000e+00       1.224647e-16       -0.991677        0.128748   
561        5.000000e-01      -8.660254e-01        0.175531       -0.984474   

      cos_semaine_annee  sin_semaine_annee  
25             0.935016           0.354605  
208           -0.354605           0.935016  
35             0.935016           0.354605  
783            0.748511           0.663123  
997           -0.568065          -0.822984  
...                 ...                ...  
512           -0.120537    

In [75]:
model = XGBRegressor(objective='reg:squarederror', 
                      max_depth=4, 
                      n_estimators=40, #first param
                      learning_rate=0.1)   #second param

In [76]:
model.fit(X_train,y_train)

XGBRegressor(base_score=0.5, booster='gbtree', colsample_bylevel=1,
             colsample_bynode=1, colsample_bytree=1, gamma=0, gpu_id=-1,
             importance_type='gain', interaction_constraints='',
             learning_rate=0.1, max_delta_step=0, max_depth=4,
             min_child_weight=1, missing=nan, monotone_constraints='()',
             n_estimators=40, n_jobs=8, num_parallel_tree=1, random_state=0,
             reg_alpha=0, reg_lambda=1, scale_pos_weight=1, subsample=1,
             tree_method='exact', validate_parameters=1, verbosity=None)

In [77]:
y_pred=model.predict(X_test)
y_pred.shape

(309,)

In [78]:
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score, max_error
import math

mse = mean_squared_error(y_test, y_pred)

rmse = math.sqrt(mse)

mae = mean_absolute_error(y_test, y_pred)

rsquared = r2_score(y_test, y_pred)

max_error = max_error(y_test, y_pred)

print('MSE =', round(mse, 2))
print('RMSE =', round(rmse, 2))
print('MAE =', round(mae, 2))
print('R2 =', round(rsquared, 2))
print('Max Error =', round(max_error, 2))

MSE = 2353391.99
RMSE = 1534.08
MAE = 990.28
R2 = 0.8
Max Error = 11611.72


MSE = 1225989.26
RMSE = 1107.24
MAE = 777.74
R2 = 0.9
Max Error = 4769.4

MSE = 2353391.99
RMSE = 1534.08
MAE = 990.28
R2 = 0.8
Max Error = 11611.72

In [72]:
from sklearn.inspection import permutation_importance



In [73]:
permutation_score = permutation_importance(model, X_train, y_train, n_repeats=100,n_jobs=-1,scoring='neg_mean_absolute_error') # Perform Permutation

importance_df = pd.DataFrame(np.vstack((X_train.columns,
                                        permutation_score.importances_mean)).T) # Unstack results
importance_df.columns=['feature','score decrease']

importance_df.sort_values(by="score decrease", ascending = False) # Order by importance

feature score decrease
1         moyen_7_services    1688.490862
3   moyenne_3der_j&service     408.637683
0                  service     391.561398
25         sin_jour_de_sem     247.561147
2        moyen_31_services     110.721084
24         cos_jour_de_sem      59.445028
12                temp_max      38.494855
30          cos_jour_annee      29.842879
27        sin_jour_du_mois      26.908269
31          sin_jour_annee      17.246517
10              feels_like      16.185018
11                temp_min      14.212477
22                    rain       11.16664
15          vacances_paris      10.296945
14              clouds_all       9.767091
9                     temp       8.932968
6             fashion_week       7.850978
13              wind_speed       7.298305
5            roland_garros       4.357916
32       cos_semaine_annee       4.243063
26        cos_jour_du_mois       3.741079
28       cos_mois_de_annee       3.150098
17                  clouds       1.039083
29       sin_mois_de_annee       0.362405
21                    mist       0.287874
33       sin_semaine_annee       0.263603
23            thunderstorm            0.0
20                     fog            0.0
19        drizzle_and_rain            0.0
18                 drizzle            0.0
16                   clear            0.0
8       match_happening_cl            0.0
7          match_happening            0.0
4                match_edf            0.0